### DISCUSSION TITLE
Upgrade from 5.7.12 to 5.7.14 fails with MySQL 1045 Error (non docker install/upgrade)

### DISCUSSION THREAD

**Discussion by dschigt** [firefly-iii/firefly-iii#6578] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hi,

In the attempt to upgrade Firefly-III to the newest version, I encounter this error upon command: "php artisan migrate --seed":

"SQLSTATE[HY000] [1045] Access denied for user 'firefly'@'localhost' (using password: YES) (SQL: select * from information_schema.tables where table_schema = firefly and table_name = migrations and table_type = 'BASE TABLE')"

Best regards,

Detlev Schigt

### Expected behaviour

What I would expect is an upgraded Firefly-III version.
To be exact: from 5.7.12 to 5.7.14

### Debug information

Unable to provide, since I am attempting to upgrade my existing installation...

Although next is shown on schren:

Illuminate\Database\QueryException

  SQLSTATE[HY000] [1045] Access denied for user 'firefly'@'localhost' (using password: YES) (SQL: select * from information_schema.tables where table_schema = firefly and table_name = migrations and table_type = 'BASE TABLE')

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:759
    755▕         // If an exception occurs when attempting to run a query, we'll format the error
    756▕         // message to include the bindings with SQL, which will make this exception a
    757▕         // lot more helpful to the developer instead of just the database's errors.
    758▕         catch (Exception $e) {
  ➜ 759▕             throw new QueryException(
    760▕                 $query, $this->prepareBindings($bindings), $e
    761▕             );
    762▕         }
    763▕     }

### Steps to reproduce

php artisan migrate --seed

### Additional info

* The upgrade steps taken are described at page: https://docs.firefly-iii.org/firefly-iii/installation/upgrade/
* It concerns a non docker install (thus I followed the steps at section: "Virtual or real server")
* The steps are executed with 'sudo', without 'sudo' and even as 'sudo -i'. All result te same
* Various issues I have walked through, none of them seem to relate to my issue. The closest I can reference is this:
https://github.com/firefly-iii/firefly-iii/issues/2284
Yet this did not yield any success...

**Comment by JC5** [comment]
> Access denied for user 'firefly'@'localhost' 

Depending on how you manage your installation, the DB_PASSWORD may have become incorrect. How is the password stored?

**Comment by dschigt** [comment]
The password is stored in the .env file:

# Use "pgsql" for PostgreSQL
# Use "mysql" for MySQL and MariaDB.
# Use "sqlite" for SQLite.
DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=firefly
DB_USERNAME=firefly
DB_PASSWORD=F******2P
# leave empty or omit when not using a socket connection
DB_SOCKET=

Whereas ‘*****’ are deleted characters of the entire password ;-)

**Reply by JC5** [reply]
Does it have a `#`-character by any chance?

**Reply by IcyBow** [reply]
Thank you a lot for this post. I just wanted to add, that I encountered exactly the same problem with the `$` sign.
Thanks again

**Comment by dschigt** [comment]
Yes….. Indeed… The password does contain such character…
Let me try it with another value… Keep you posted….

**Reply by JC5** [reply]
Either try `"pass#word"` or avoid the #.

**Comment by dschigt** [comment]
Thx! That did the trick ;-)
In the meantime I have upgraded from 5.7.12 to 5.7.14.

Thanks again and have a great day, James!
